UefiPayloadPkg/UniversalPayloadBuild.py

```diff
@@ -140,6 +139,11 @@ def BuildUniversalPayload(Args):
     else:
         print("Incorrect arch option provided")
 
+    if "X64" in BuildArch:
+        BuildDir   = os.path.join(os.environ['WORKSPACE'], os.path.normpath("Build/UefiPayloadPkgX64"))

```

Comment:
looks outcome is the same as original code, what was the issue this patch is fixing?

Reply:
If `Args.Arch` is set to "IA32" then we had a build error, because it still tried to read files out of "Build/UefiPayloadPkgX64". This might just be related to https://github.com/tianocore/edk2/blob/11d4edc7c6dadd4992cb75e4581de5867590e694/UefiPayloadPkg/UefiPayloadPkg.dsc#L25, and we could fix it by passing `-D BUILD_ARCH=IA32` if so.

Reply:
Could you share how to reproduce the build failure? I tried "py UniversalPayloadBuild.py -a IA32" and the build is successful.

Reply:
As I said offline, the issue was that coreboot was overriding BUILD_ARCH=X64 for all EDK2 payloads, which is incorrect for UPL. I've since fixed this.